Types are not correctly associated with deep file imports

If you import top-level functions from `hyparquet` then everything is fine.

If you try to import from files within a package, which is legal javascript, the javascript __works__ but typescript complains that it failed  to find our generated types:

```js
import { readZigZag } from 'hyparquet/src/thrift.js'
``` 

```
$ tsc
src/index.js:4:28 - error TS7016: Could not find a declaration file for module 'hyparquet/src/thrift.js'. '/home/platypii/code/libs/hyparquet-import/node_modules/hyparquet/src/thrift.js' implicitly has an 'any' type.
  If the 'hyparquet' package actually exposes this module, try adding a new declaration (.d.ts) file containing `declare module 'hyparquet/src/thrift.js';`

4 import { readZigZag } from 'hyparquet/src/thrift.js'
```

For whatever reason it's not picking up our generated types in the `types` directory.

--- a/package.json
+++ b/package.json
@@ -24,6 +24,16 @@
   ],
   "type": "module",
   "types": "types/hyparquet.d.ts",
+  "exports": {
+    ".": {
+      "import": "./src/hyparquet.js",
+      "types": "./types/hyparquet.d.ts"
+    },
+    "./src/*.js": {
+      "import": "./src/*.js",
+      "types": "./types/*.d.ts"
+    }
+  },
   "scripts": {
     "build:types": "tsc -p ./tsconfig.build.json",
     "coverage": "vitest run --coverage --coverage.include=src",

